static/client/services/api/hooks/products.ts

```diff
@@ -0,0 +1,19 @@
+import { useQuery } from "react-query";
+
+import { ProductsServices } from "@/services/api/services/products";
+import type { IProductsResponse } from "@/services/api/types/products";
+import type { IApiBasicError, IUseQueryHookRest } from "@/services/api/types/query";
+
+export function useProducts(): IUseQueryHookRest<IProductsResponse> {
+  const result = useQuery<IProductsResponse, IApiBasicError>({
+    queryKey: "products",
+    queryFn: ProductsServices.getProducts,
+  });
+
+  const isLoading = result.isLoading;
+  const isFetching = result.isFetching;
```

Comment:
you usually need one or another (either isLoading or isFetching). There was only one use case when both were needed, but that's it. In your case looks like you don't need both, then why use memory for storing them?